### PR TITLE
Modify mock vsts-task-lib filter function to return a function

### PR DESCRIPTION
**Pull request by jotaylo**
The filter function in minimatch returns a function rather than a string[], this change modifies the filter function in the mocked task lib to also return a function.

None of the tests in master currently specify filters in their configuration files.

### CODE REVIEW

File: Tests/lib/vsts-task-lib/task.ts [hunk 1]
```diff
@@ -666,9 +666,12 @@ export function matchFile(list, pattern, options): string[] {
     return mock.getResponse('match', pattern) || [];
 }
 
-export function filter(pattern, options): string[] {
-    return mock.getResponse('filter', pattern) || [];
-}    
+export function filter(pattern, options): any {
+	var filterList = mock.getResponse('filter', pattern) || [];
+	return function(pattern, i, filterList) {
+		return filterList;
```
Comment (ericsciple) [inline Tests/lib/vsts-task-lib/task.ts]: Shouldn't this nested function just return `filterList[i]`?

Comment (ericsciple) [inline Tests/lib/vsts-task-lib/task.ts]: If you use .filter, then should your answers be an array of booleans. It's confusing. Would it be simpler to use tl.match instead?

Comment (ericsciple) [inline Tests/lib/vsts-task-lib/task.ts]: (i was wrong the first time, the inner function should not return `filterList[i]`, it should return a bool)

**Comment by jotaylo** [comment]
Modified it so that it actually filters now.

File: Tests/lib/vsts-task-lib/task.ts [hunk 2]
```diff
@@ -666,9 +666,12 @@ export function matchFile(list, pattern, options): string[] {
...
 }
 
-export function filter(pattern, options): string[] {
-    return mock.getResponse('filter', pattern) || [];
-}    
+export function filter(pattern, options): any {
+	var filterList = mock.getResponse('filter', pattern) || [];
+	return function(pattern, i, arr) {
+		return filterList.indexOf(pattern) >= 0;
+	}
```
Comment (ericsciple) [inline Tests/lib/vsts-task-lib/task.ts]: seems appropriate. kind of quirky, but ultimately a better choice i think over an array of booleans

**Pull request merged**